Feature/s1 6.1 backend fixes updates

gdeak-unosquare opened the pull request:
- A couple of commits included

gdeak-fmi committed:
S1-6 Backend fixes/updates
- Exposed port 5432 for postgres in docker-compose.yaml
- Update postgres config to hit localhost (postgres docker container can be used)
- Added AppConfig.java - spring configuration for automapper bean
- Added SignUpRequestDto.java - this need modified
- Autowired in automapper bean
- Updated application.properties postgres to hit localhost postgres

gdeak-fmi committed:
- S1.6.6 - Flyway migrattion was not working, this commit should fix the problem
- It looks like this triggered a docs

gdeak-unosquare committed:
S1.6.6

Added missing dependency to run flyway migrations automatically

gdeak-fmi committed:
Merge remote-tracking branch 'remotes/origin/develop' into Feature/s1-6.1-backend-fixes-updates

gdeak-unosquare committed:
Added iml intellij file to gitignore

gdeak-fmi committed:
Merge remote-tracking branch 'remotes/origin/develop' into Feature/s1-6.1-backend-fixes-updates

gdeak-fmi committed:
S1.6.3 Check/update schema
- Updated migration sql script to include missing tables
- Added migration scripts to include some of the required data

gdeak-unosquare committed:
S1.6.3 Check/update schema
- Added employee status (populated with values)
S1.6.7 Use DTOs - register user not working correctly
- User registration works as expected now

gdeak-unosquare committed:
S1-6
- Added mockito-core dependency
- Removed unnecessary @JsonProperty annotations
- Updated/fixed unit tests

BrownMark commented on BackEnd/src/main/java/com/unosquare/admin_core/back_end/enums/converter/ClientStatusConverter.java:
```diff
@@ -7,9 +7,9 @@
 @Converter
 public class ClientStatusConverter
-        implements AttributeConverter<ClientStatus, Short> {
+        implements AttributeConverter<ClientStatus, Integer> {
```
I don't think we actually use these converters anymore.

gdeak-unosquare replied:
Yeah, I am planning to remove those soon, wanted to put the PR in. That is what some of the work is to cleanup some of this stuff.

BrownMark commented on BackEnd/src/main/java/com/unosquare/admin_core/back_end/enums/converter/ContractStatusConverter.java:
```diff
@@ -7,17 +7,17 @@
 @Converter
 public class ContractStatusConverter
-        implements AttributeConverter<ContractStatus, Short> {
+        implements AttributeConverter<ContractStatus, Integer> {
```
Same again, do we actually use this?

gdeak-unosquare replied:
Same, I am planning to remove those soon, wanted to put the PR in. That is what some of the work is to cleanup some of this stuff.

BrownMark commented on BackEnd/src/main/java/com/unosquare/admin_core/back_end/enums/converter/CountryConverter.java:
```diff
@@ -7,17 +7,17 @@
+        implements AttributeConverter<Country, Integer> {
 
-    public Short convertToDatabaseColumn(Country country) {
+    public Integer convertToDatabaseColumn(Country country) {
```
Same again, do we actually use this?

gdeak-unosquare replied:
Same, I am planning to remove those soon, wanted to put the PR in. That is what some of the work is to cleanup some of this stuff.

BrownMark commented on BackEnd/src/main/java/com/unosquare/admin_core/back_end/dto/ContractDto.java:
```diff
@@ -32,7 +32,7 @@ public ContractDto() {
     public ContractDto(int clientId, int employeeId, int contractStatusId) {
         this.client = new ModelMapper().map(clientService.findById(clientId), ClientDto.class);
         this.employee = new ModelMapper().map(employeeService.findById(employeeId), EmployeeDto.class);
```
We can wire up the model mapper here. I'm not sure if we want to be using the services etc inside the DTO though.

gdeak-unosquare replied:
Yeah, did not get to this one yet :) We should not create instances like this. Anyway, that constructor is not even used, so ignore that for the moment, it will probably go away soon.

gdeak-unosquare committed:
S1-6
- Removed unused converters: converter/CountryConverter.java, converter/EmployeeRoleConverter.java and converter/EmployeeStatusConverter.java

gdeak-unosquare commented:
Removed unused converters, some converters are still used in some parts of the code

BrownMark reviewed (approved):
Looks good. Good work Gabe.
